src/mod_muc.erl

```diff
@@ -246,11 +246,42 @@ config_spec() ->
                  <<"hibernated_room_timeout">> => #option{type = int_or_infinity,
                                                           validate = non_negative},
                  <<"default_room">> => default_room_config_spec()
-                }
+                },
+       defaults = defaults()
       }.
 
+defaults() ->
+    #{<<"backend">> => mnesia,
+      <<"host">> => default_host(),
+      <<"access">> => all,
+      <<"access_create">> => all,
+      <<"access_admin">> => none,
+      <<"access_persistent">> => all,
+      <<"history_size">> => 20,
+      <<"room_shaper">> => none,
+      <<"max_room_id">> => infinity,
+      <<"max_room_name">> => infinity,
+      <<"max_room_desc">> => infinity,
+      <<"min_message_interval">> => 0,
+      <<"min_presence_interval">> => 0,
+      <<"max_users">> => ?MAX_USERS_DEFAULT,
+      <<"max_users_admin_threshold">> => 5,
+      <<"user_message_shaper">> => none,
+      <<"user_presence_shaper">> => none,
+      <<"max_user_conferences">> => 10,
+      <<"http_auth_pool">> => none,
+      <<"load_permanent_rooms_at_startup">> => false,
+      <<"hibernate_timeout">> => timer:seconds(90),
+      <<"hibernated_room_check_interval">> => infinity,
+      <<"hibernated_room_timeout">> => infinity,
+      <<"default_room">> => keys_as_atoms(default_room_opts())}.
+
+keys_as_atoms(Map) ->
+    maps:from_list([{binary_to_atom(K, latin1), V} || {K, V} <- maps:to_list(Map)]).
```

Comment:
Minor: I would drop the `latin1` argument because by default it uses `utf8` and the names are ascii anyway.

Reply:
omg, it works. I remember times when there was only binary_to_atom/2 :D